File: interview/1.1/main11.py
def unique_char(s):
    if len(s) > 128:
        return False
    char_set = [False] * 128
    for c in s:
        char_int = ord(c) - ord('A')
        print(char_int)
        if char_set[char_int] == False:
            char_set[char_int] = True
        else:
            return False
    return True

print(unique_char('abced'))
print(unique_char('abceda'))
print(unique_char('abcedA'))

File: interview/1.1/Main11.java
public class Main11 {
    // ASCII文字範囲での判定メソッド
    public static boolean isUniqueChar(String str) {
        if(str.length() > 128) return false;
        
        boolean[] char_set = new boolean[128];

        for(int i = 0; i < str.length(); i++) {
            int char_int = str.charAt(i) - 'A';
            if(char_set[char_int] == false) {
                char_set[char_int] = true;
            } else {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        System.out.println(isUniqueChar(args[0]));
    }
}